src/Audio.md

```diff
@@ -0,0 +1,110 @@
+This also means that the APU runs about 2.4% faster on the SGB1, increasing frequencies by as much and thus sounding slightly higher-pitched.
+The SGB2 rectifies this issue.
+
+The APU processes a lot of events at a fixed rate dependent on the event's type.
+All interfaces to the APU use **durations** instead of frequencies, which may be confusing as signal theory and music are more typically based on the latter.
+Thus, durations will be expressed from their frequencies: for example, a "256 Hz tick" means "1 ∕ 256th of a second".
+
+The length of APU ticks is not affected by [CGB double speed](<#FF4D — KEY1 (CGB Mode only): Prepare speed switch>), so the APU works just the same regardless of CPU speed.
+
+::: warning
+
+The Game Boy's APU is actually full of tricky details; this chapter will mostly describe the intended / common behavior, and often paper over bugs & quirks.
+Readers wishing to learn more should read the [APU details](<#Audio Details>) chapter.
+
+:::
+
+### Triggering
+
+**Triggering** a channel causes it to turn on if it wasn't[^trig_dac_off], and to start playing its wave from the beginning[^pulse_restart].
+Most changes to a channel's parameters take effect immediately, but some require re-triggering the channel.
+
+### Volume & envelope
+
+The volume can be controlled in two ways: there is a "master volume" control[^vol_knob] (which has separate settings for the left and right outputs), and each channel's volume can be individually set as well (CH3's less precisely than the others).
+
+Additionally, an [**envelope**](https://en.wikipedia.org/wiki/Envelope_(music)) can be configured for all channels except CH3, which allows automatically adjusting the volume over time.
+The parameters that can be controlled are how many times the envelope will be stepped, in which direction, and how long between steps.
+All envelopes are ticked at 64 Hz, and every 1–7 of those ticks, the volume will be increased or decreased.
+
+### Length timer
+
+All channels can be individually set to automatically shut themselves down after a certain amount of time.
+
+If the functionality is enabled, a channel's **length timer** ticks up[^len_cnt_dir] at 256 Hz (tied to [DIV-APU](<#DIV-APU>)) from the value it's initially set at.
+When the length timer reaches 64, the channel is turned off.
+
+### Frequency
+
+Music notes and audio waves are typically manipulated in terms of **frequency**[^pitch], i.e. how often the signal repeats per second.
+However, as explained above, the Game Boy APU primarily works with durations; thus, **wavelengths** will be used instead of frequency.[^len_raw]
+
+::: warning
+
+The term "wavelength" throughout this document does *not* designate the inverse of the frequency, but instead a quantity akin to it.
+See the description of each NRx3 register for more information.
+
+:::
+
+---
+
+[^trig_dac_off]:
+If [the channel's DAC](#DACs) is off, the channel will not turn on.
```

Comment:
This footnote makes little sense in the context it's linked from. "The channel will not turn on" in which situation? I think what this means to say is that the channel doesn't initially turn on just by writing other registers than bit 7 of the NRx4 but this doesn't come across.

Reply:
I meant that setting NRx4 bit 7 while the DAC is still off will not turn the channel on. The DAC must be turned on first, then the channel triggered.

Reply:
Ok, I have to admit I was unaware of this particular fact: 
> Channel <var>x</var>'s DAC is enabled if and only if `[NRx2] & $F8 != 0`;"
I was under the impression that this was not a prerequisite for turning on the channel (in terms of for example making it's phase counter start counting). Is the research for this behavior documented anywhere?

Reply:
I don't know of any test ROM that checks this (maybe somewhere in blargg), but I just tested with [Numism's exerciser](https://github.com/pinobatch/numism/tree/main/gameboy/exercise#sound-exerciser), writing $80 to NR24 does nothing with NR22 = $00, but it does with e.g. $08.